### PR TITLE
Docs second wave

### LINKED ISSUE #428: Document all APIs
**Is your feature request related to a problem? Please describe.**
There is no concise place where all apis are documented for Tempo.  We should add this.  For the various ingestion protocols we should just link to external docs.  There's no reason to replicate it.

### PR DESCRIPTION
**Pull request by annanay25**
Changes:
- Add API documentation. Fixes #428 
- Fix home page links
- Change mixin `out` folder to `yamls` and add supporting documentation on the README
- Combine two sections in troubleshooting that were very similar
- Move `pushing-spans-with-http.md` under API documentation
- Add agent troubleshooting section

Signed-off-by: Annanay <annanayagarwal@gmail.com>

### CODE REVIEW

File: docs/tempo/website/troubleshooting/unable-to-see-trace.md [hunk 1]
```diff
@@ -52,7 +57,36 @@ This can also be confirmed by checking the metric `tempo_request_duration_second
...
+
+If you are using the Grafana Agent, continue reading the following section for metrics to monitor.
+
+### Diagnosing the issue
+Check if the pipeline is dropping spans. The following metrics on the Grafana Agent help determine this -
+- `tempo_exporter_send_failed_spans`. The value of this metric should be 0.
+- `tempo_receiver_refused_spans`. This value of this metric should be 0.
+- `tempo_processor_dropped_spans`. The value of this metric should be 0.
+
+If the pipeline is not reporting any dropped spans, check whether application spans are being dropped by Tempo. The following metrics help determine this -
```
Comment (mapno) [inline docs/tempo/website/troubleshooting/unable-to-see-trace.md:76]: I think this is not very clear. `tempo_receiver_refused_spans` represents the number of spans dropped by Tempo and the Grafana Agent. I think it's worth explaining that both share the same metric.

**Review by mapno — COMMENTED** [review]
Nice! LGTM overall

File: docs/tempo/website/community/_index.md [hunk 2]
```diff
@@ -3,7 +3,13 @@ title: Community
 weight: 600
 ---
 
-# Contribute
+## Communicate
+
+- [Grafana Slack](https://slack.grafana.com/) #tempo channel
+- [Grafana Tempo Community Forum](https://community.grafana.com/c/grafana-tempo/40)
+- [Google Groups](https://groups.google.com/forum/#!forum/tempo-users)
```
Comment (joe-elliott) [inline docs/tempo/website/community/_index.md:10]: Add something in here about the community call please. 

Link to the google doc: https://docs.google.com/document/d/1yGsI6ywU-PxZBjmq3p3vAXr9g5yBXSDk4NU8LGo8qeY/edit#

File: docs/tempo/website/api_docs/_index.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
+---
+title: API documentation
```
Comment (joe-elliott) [inline docs/tempo/website/api_docs/_index.md:2]: Missing endpoints:
/api/echo
/compactor/ring
/distributor/ring
/flush

File: docs/tempo/website/api_docs/_index.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+| API | Service | Type | Endpoint |
+| --- | ------- | ---- | -------- |
+| [Configuration](#configuration) | _All services_ |  HTTP | `GET /config` |
+| [Readiness probe](#readiness-probe) | _All services_ |  HTTP | `GET /ready` |
+| [Metrics](#metrics) | _All services_ |  HTTP | `GET /metrics` |
+| [Pprof](#pprof) | _All services_ |  HTTP | `GET /debug/pprof` |
+| [Ingest traces](#ingest) | Distributor |  - | See section for details |
+| [Querying traces](#query) | Query-frontend |  HTTP | `GET /api/traces/<traceID>` |
+| [Memberlist](#memberlist) | Distributor, Ingester, Querier, Compactor |  HTTP | `GET /memberlist` |
+| [Shutdown](#shutdown) | Ingester |  HTTP | `GET,POST /ingester/shutdown` |
```
Comment (joe-elliott) [inline docs/tempo/website/api_docs/_index.md:25]: path should just be `/shutdown`

**Review by joe-elliott — APPROVED** [review]
A few small changes, but overall great work!

**Pull request merged**